spectrochempy/analysis/_base.py

```diff
@@ -1043,6 +1085,433 @@ def Y(self):
+            Reduced `X` data, where `n_observations` is the number of observations
+            and `n_components` is the number of components. If `X_transform` is not
+            provided, a transform of `X` provided in `fit` is performed first.
+        %(kwargs)s
+
+        Returns
+        -------
+        `~spectrochempy.core.dataset.nddataset.NDDataset`
+            Dataset with shape (:term:`n_observations`\ , :term:`n_features`\ ).
+
+        Other Parameters
+        ----------------
+        %(analysis_transform.other_parameters)s
+
+        See Also
+        --------
+        reconstruct : Alias of inverse_transform (Deprecated).
+        """
+        if not self._fitted:
+            raise NotFittedError
+
+        if isinstance(X_transform, NDDataset):
+            X_transform = X_transform.data
+
+        elif X_transform is None:
+            X_transform = self.transform(**kwargs).data
+
+        if isinstance(Y_transform, NDDataset):
+            Y_transform = Y_transform.data
+
+        elif Y_transform is None and both is True:
+            Y_transform = self.transform(**kwargs).data
+
+        if Y_transform is None:
+            X = self._inverse_transform(X_transform)
+            return X
+        else:
+            X, Y = self._inverse_transform(X_transform, X_transform)
+            return X, Y
+
+    _docstring.get_sections(
+        _docstring.dedent(inverse_transform.__doc__),
+        base="analysis_inverse_transform",
+        sections=["Parameters", "Returns"],
+    )
+    # _docstring.keep_params("analysis_inverse_transform.parameters", "X_transform")
+
+    @_docstring.dedent
+    def fit_transform(self, X, Y, both=False):
+        """
+        Fit the model with `X` and `Y` and apply the dimensionality reduction on `X`
+        and optionally on `Y`\ .
+
+        Parameters
+        ----------
+        %(analysis_fit.parameters.X)s
+        %(analysis_fit.parameters.Y)s
+        both: bool, optional
+            Whether to apply the dimensionality reduction on `X` and `Y` .
+        %(kwargs)s
+
+        Returns
+        -------
+        %(analysis_transform.returns)s
+
+        Other Parameters
+        ----------------
+        %(analysis_transform.other_parameters)s
+        """
+
+        self.fit(X, Y)
+        if both:
+            return self.transform(X, Y)
+        else:
+            return self.transform(X)
+
+    # ----------------------------------------------------------------------------------
+    # Plot methods
+    # ----------------------------------------------------------------------------------
+    @_docstring.dedent
+    def parityplot(
+        self,
+        Y=None,
+        Y_hat=None,
+        clear=True,
+        s=None,
+        c=None,
+        marker=None,
+        cmap=None,
+        norm=None,
+        vmin=None,
+        vmax=None,
+        alpha=0.5,
+        linewidths=None,
+        edgecolors=None,
+        plotnonfinite=False,
+        data=None,
+        **kwargs,
```

Comment:
Why not put non essential parameters into **kwargs?

Reply:
I have not really finished to review. For the moment it is mainly for cosmetics and documentation rendering. 

I will now look at the code change more deeply.